packages/flutter_deck/lib/src/controls/flutter_deck_controls.dart

```diff
@@ -468,22 +468,31 @@ class _FullscreenButton extends StatelessWidget {
     final controlsNotifier = flutterDeck.controlsNotifier;
     final markerNotifier = flutterDeck.markerNotifier;
 
-    final isInFullscreen = controlsNotifier.isInFullscreen();
-    final onPressed = isInFullscreen
-        ? controlsNotifier.leaveFullscreen
-        : controlsNotifier.enterFullscreen;
-
-    return ListenableBuilder(
-      listenable: markerNotifier,
-      builder: (context, _) => MenuItemButton(
-        leadingIcon: Icon(
-          isInFullscreen
-              ? Icons.fullscreen_exit_rounded
-              : Icons.fullscreen_rounded,
-        ),
-        onPressed: !markerNotifier.enabled ? onPressed : null,
-        child: Text(isInFullscreen ? 'Leave full screen' : 'Enter full screen'),
-      ),
+    return FutureBuilder(
+      future: controlsNotifier.isInFullscreen(),
+      initialData: false,
+      builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {
+        final isInFullscreen = snapshot.data ?? false;
+        final onPressed = isInFullscreen
+            ? controlsNotifier.leaveFullscreen
+            : controlsNotifier.enterFullscreen;
+        return ListenableBuilder(
+          listenable: markerNotifier,
+          builder: (context, _) => MenuItemButton(
+            leadingIcon: Icon(
+              isInFullscreen
+                  ? Icons.fullscreen_exit_rounded
+                  : Icons.fullscreen_rounded,
+            ),
+            onPressed: !markerNotifier.enabled ? onPressed : null,
+            child: Text(
+              isInFullscreen //
```

Comment:
Redundant comment.

Reply:
This comment is to better format the code into multiple lines. But if you do not like the formatting I can remove it

Reply:
Ok, that's a neat trick. We simply rely on default formatting in the project, that's where this comment comes from.